code/modules/mob/living/simple_animal/vore/otie.dm

```diff
@@ -0,0 +1,297 @@
+				T = M
+				break
+
+	if(T) //Permission to fuck up and vore GET!
+		ai_log("FindTarget() found [T]!",1)
+		if(set_target(T))
+			handle_stance(STANCE_ATTACK)
+
+	return T
+
+/mob/living/simple_animal/otie/fluff/security/FindTarget()
+	var/atom/T = null
+	for(var/atom/A in ListTargets(view_range))
+
+		if(A == src)
+			continue
+
+		var/atom/F = Found(A)
+		if(F)
+			T = F
+			break
+		else if(specific_targets)
+			return 0
+
+		if(isliving(A))
+			var/mob/living/L = A
+			if(L.faction == src.faction && !attack_same)
+				continue
+			else if(L in friends)
+				continue
+			else if(!SA_attackable(L))
+				continue
+			else if(tamed == 1 && ishuman(L))
+				continue
+			else if(tamed == 1 && isrobot(L))
+				continue
+			else
+				T = L
+				break
+
+		else if(istype(A, /obj/mecha)) // Our line of sight stuff was already done in ListTargets().
+			var/obj/mecha/M = A
+			if(!SA_attackable(M))
+				continue
+			if((M.occupant.faction != src.faction) || attack_same)
+				T = M
+				break
+
+	if(T) //Permission to fuck up and vore GET!
+		ai_log("FindTarget() found [T]!",1)
+		if(set_target(T))
+			if(check_threat(T) >= 4)
+				broadcast_security_hud_message("[src] is attempting to 'detain' suspect <b>[target_name(T)]</b> in <b>[get_area(src)]</b>.", src)
+				handle_stance(STANCE_ATTACK)
+			else
+				handle_stance(STANCE_ATTACK)
+
+	return T
+
+/mob/living/simple_animal/otie/fluff/security/proc/target_name(mob/living/T)
+	if(ishuman(T))
+		var/mob/living/carbon/human/H = T
+		return H.get_id_name("unidentified person")
+	return "unidentified lifeform"
+
+//Basic friend AI
+
+/mob/living/simple_animal/otie/fluff/Life()
+	. = ..()
+	if(!. || ai_inactive || !friend) return
+
+	var/friend_dist = get_dist(src,friend)
+
+	if (friend_dist <= 4)
+		if(stance == STANCE_IDLE)
+			if(set_follow(friend))
+				handle_stance(STANCE_FOLLOW)
+				if(resting)
+					lay_down()
+
+	if (friend_dist <= 1)
+		if (friend.stat >= DEAD || friend.health <= config.health_threshold_softcrit)
+			if (prob((friend.stat < DEAD)? 50 : 15))
+				var/verb = pick("whines", "yelps", "whimpers")
+				audible_emote(pick("[verb] in distress.", "[verb] anxiously."))
+		else
+			if (prob(5))
+				visible_emote(pick("nuzzles [friend].",
+								   "brushes against [friend].",
+								   "rubs against [friend].",
+								   "noses softly at [friend].",
+								   "slobberlicks [friend].",
+								   "murrs."))
+	else if (friend.health <= 50)
+		if (prob(10))
+			var/verb = pick("whines", "yelps", "whimpers")
+			audible_emote("[verb] anxiously.")
+
+/mob/living/simple_animal/otie/Life()
+	. = ..()
+	if(!. || ai_inactive) return
+
+	if(prob(5))
+		lay_down()
+
+//Pet 4 frond
+
+/mob/living/simple_animal/otie/attack_hand(mob/living/carbon/human/M as mob)
+	..()
+	switch(M.a_intent)
+
+		if(I_HELP)
+			if (health > 0)
+				LoseTarget()
+				handle_stance(STANCE_IDLE)
+				if(prob(50)) //It's a fiddy-fiddy you may get a buddy pal or you may get mauled and ate. Win-win!
```

Comment:
This will not result in you being mauled and eaten since the parent also checks intent, or maybe that was just a joke. :P Or did you mean WHILE BEING ATTACKED and patting them?